streams/src/main/java/org/apache/kafka/streams/processor/internals/RepartitionTopics.java

```diff
@@ -119,7 +119,8 @@ public Queue<StreamsException> missingSourceTopicExceptions() {
             return new StreamsException(
                 new MissingSourceTopicException(String.format(
                     "Missing source topics %s for subtopology %d of topology %s",
-                    missingSourceTopics, subtopologyId, topologyName)),
+                    missingSourceTopics, subtopologyId, topologyName),
+                    missingSourceTopics),
```

Comment:
Would it be possible to return a set of missing source topics from `RepartitionTopics`?

Reply:
FYI, i write skeleton code below! 
Does it make sense to you as well?
```java
public class RepartitionTopics {
    ...
    // Add new field (private)
    private final Set<String> missingTopics = new HashSet(); 

    ...
    public Queue<StreamsException> missingSourceTopicExceptions() {...}

    // Add new method (package-private)
    Set<String> getMissingTopics() {
      return this.missingTopics;
    }
   ...
}
```

Reply:
The public interface is defined as everything that shows up in the [javadocs](https://kafka.apache.org/37/javadoc/). Classes in a package whose name contains `internals` does not show up in the javadocs. Class `RepartitionTopics` is in package `org.apache.kafka.streams.processor.internals`. Thus, `RepartitionTopics` is not part of the public interface.
Regarding field `missingTopics`, there is already `missingInputTopicsBySubtopology` that includes all missing topics.

Reply:
I really appreciate your kind explanation. 👍 
I've understood it clearly, and thanks to you, I've established the correct direction for revision.